src/core/immerClass.ts

```diff
@@ -177,9 +177,6 @@ export class Immer implements ProducersFns {
 	 * By default, feature detection is used, so calling this is rarely necessary.
 	 */
 	setUseProxies(value: boolean) {
-		if (!hasProxies) {
```

Comment:
could you change this to `if (value && !hasProxies)` ?

Reply:
ok, it make sense

Reply:
> could you change this to `if (value && !hasProxies)` ?

I just submit a new commit